src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h

```diff
@@ -494,6 +502,52 @@ namespace OpenMS
 
     /// Find largest peak in a vector of chromatograms
     void findLargestPeak(std::vector<MSChromatogram >& picked_chroms, int& chr_idx, int& peak_idx);
+    
+    /**
+    @brief Will use the chromatogram to get the maximum peak intensity
+
+    The maximum peak intensity/height is calculated.  The convex hull points,
+    intensity_sum, and rt_sum are also calculated.
+    */
+    void calculatePeakApexInt_(const MSChromatogram& chromatogram,
+    double best_left, double best_right, 
+    ConvexHull2D::PointArrayType & hull_points,
+    double & intensity_sum, 
+    double & intensity_integral,
+    double & rt_sum,
+    double & peak_apex_int,
+    double & peak_apex_rt);
+
+    // internal structure to represent various peak shape metrics
+    struct PeakShapeMetrics_ {
+      double width_at_5 = 0.0;
+      double width_at_10 = 0.0;
+      double width_at_50 = 0.0;
+      double start_time_at_10 = 0.0;
+      double start_time_at_5 = 0.0;
+      double start_time_at_50 = 0.0;
+      double end_time_at_10 = 0.0;
+      double end_time_at_5 = 0.0;
+      double end_time_at_50 = 0.0;
+      double total_width = 0.0;
+      double tailing_factor = 0.0;
+      double asymmetry_factor = 0.0;
+      double baseline_delta_2_height = 0.0;
+      double slope_of_baseline = 0.0;
+      int points_across_baseline = 0;
+      int points_across_half_height = 0;
+    };
+
+    /**
+    @brief Calculates standard peak shape quality metrics
+
+    Standard peak shape quality metrics are calculated for down stream QC/QA.
```

Comment:
most of these are self-explanatory. can you add some comments especially for the non-standard metrics such as tailing / asymmetry factor / slope of baseline

- what is their meaning
- how are they calculated?